src/mjolnir/graphtilebuilder.cc

```diff
@@ -1319,14 +1319,16 @@ void GraphTileBuilder::AddLandmark(const GraphId& edge_id, const Landmark& landm
   }
 
   // update edgeinfo_offset_map
+  std::unordered_map<uint32_t, EdgeInfoBuilder*> new_edgeinfo_offset_map_{};
+
   for (auto& e : edgeinfo_offset_map_) {
     if (e.first > original_offset) {
-      auto* eib = e.second;
-      uint32_t newkey = e.first + shift;
-      edgeinfo_offset_map_.erase(e.first);
-      edgeinfo_offset_map_.insert(std::make_pair(newkey, eib));
+      new_edgeinfo_offset_map_.insert(std::make_pair(e.first + shift, e.second));
+    } else {
+      new_edgeinfo_offset_map_.insert(e);
     }
   }
+  edgeinfo_offset_map_ = new_edgeinfo_offset_map_;
```

Comment:
@kevinkreiser @nilsnolde here is the edited part to update edgeinfo_offset_map. the previous code manipulated the map while reading it, which would cause segmentation fault.
I guess here is still optimization to do.

Reply:
this doesnt look right to me. anything that was in there before is completely gone and will only contain the stuff you just added right?

looking at the old code i cannot understand why there would be a segfault, can you say more about what you were seeing with the old code?

Reply:
ok after spending 5 minutes looking at it while not doing something else i can see the issue. we were iterating over it while updating it which will be a problem depending on what happens with reallocations. i would have thought it wouldnt do reallocations because we arent actually growing the size but it must be since you hit a segfault.

anyway the change here will work. i think in the long term we can optimize this another way.